woocommerce/woocommerce
# Add track events for the background removal UI

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



This PR adds the following track events for the image background removal UI:

| Event  | Description |
| ------------- | ------------- |
| wcadmin_woo_ai_product_image_background_removal_view_ui  | When the remove background button is viewed in the UI.  |
| wcadmin_woo_ai_product_image_background_removal_click  | When the remove background button is clicked. |
| wcadmin_woo_ai_product_image_background_removal_complete  | When the APIs have been called successfully, and the image background removal process has been completed (after the new image is uploaded to the media library).  |
| wcadmin_woo_ai_product_image_background_removal_error  | When an error occurred while calling the background. removal API or when uploading the image to the user's media library. |

Closes #39731.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Run `localStorage.setItem( 'debug', 'wc-admin:*' );` in the browser console to set debug for tracks
2. Select a featured image for the product in the product editor to see the "Remove background" button in the media library.
3. Confirm that the track events mentioned above are logged.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
